flink-dist/src/main/flink-bin/bin/flink-console.sh

```diff
@@ -70,4 +97,10 @@ if [[ ${JAVA_VERSION} =~ ${IS_NUMBER} ]]; then
 fi
 
 echo "Starting $SERVICE as a console application on host $HOSTNAME."
+
+echo $$ >> "$pid" 2>/dev/null
```

Comment:
please add a comment for what this does

Reply:
I will add a comments here.

Reply:
so `$$` returns the PID of the current process?

> and then replace the current program with specified SERVICE in the same process

With this you mean the exec call? if so, I'd omit it, because it raises more questions than it answers. (most notably due to describing behavior several lines down)

Reply:
Yes, `$$` will return the pid of current process. For the `exec`, i will remove the comments about it.